src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -384,3 +352,98 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 
 	return FinalizePlan(planContext->plan, distributedPlan);
 }
+
+
+/*
+ * ShardPlacementWhenColocatedWithDistTable decides on a placement
+ * for delegating a procedure call that accesses a distributed table.
+ */
+static ShardPlacement *
+ShardPlacementWhenColocatedWithDistTable(DistObjectCacheEntry *procedure,
```

Comment:
it could be nice to remove the duplication between this function and `ShardPlacementWhenColocatedWithDistTable` in `call.c`. It is a bit confusing to use the same function name in two different files though, do you think we can change them to be more specific so that they are not the same

Reply:
merged them into `ShardPlacementForFunctionColocatedWithReferenceTable`